server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java

```diff
@@ -789,8 +795,66 @@ private void mergeMetadataRecords(MergeInfo info) throws AccumuloException {
       for (Entry<Key,Value> entry : scanner) {
         Key key = entry.getKey();
         Value value = entry.getValue();
+
+        final KeyExtent keyExtent = KeyExtent.fromMetaRow(key.getRow());
+
+        // Keep track of the last Key Extent seen so we can use it to fence
+        // of RFiles when merging the metadata
+        if (lastRow != null && !keyExtent.equals(lastRow)) {
+          previousKeyExtent = lastRow;
+        }
+
+        // Special case for now to handle the highest/stop tablet which is where files are
+        // merged to so we need to handle the deletes on update here as it won't be handled later
+        // TODO: Can this be re-written to not have a special edge case and make it simpler?
+        if (keyExtent.equals(stopExtent)) {
+          if (previousKeyExtent != null
+              && key.getColumnFamily().equals(DataFileColumnFamily.NAME)) {
+
+            // Delete exiting metadata as we are now adding a range
+            m.putDelete(key.getColumnFamily(), key.getColumnQualifier());
```

Comment:
May not want to always do this delete.   If the existing files range falls completely within this new tablets range then its range will not change.  If its range does not change then will its new and old metadata entries be identical?  If so maybe we only bother with the delete and add if the computed fenced range is not equal to the files existing range.

Reply:
I will check on that, but yeah in theory we don't need to re-write or change things if the range isn't different or is still valid. I made a comment about optimizations and I figured there would certainly be things to optimize or improve and this might be one of them.

Reply:
If a mutation has an add and delete for the same key, it may ultimately be deleted.  So it could be more than an optimization, may be needed for correctness but not completely sure.  It depends on what timestamp each entry from the mutation gets on the server side.  When two keys have the same timestamp and one is a delete, then the delete will sort first hiding the non delete.

Reply:
I changed this so it will only delete if the new metadata is different than before